### PR TITLE
drivers: lte_link_control: Config for APN and legacy PCO mode.

### PR DESCRIPTION
**Pull request by joakimtoe**
Adding support for
- Configuring legacy PCO mode, needed for some NB-IoT networks to get DNS working.
- Configuration for setting data APN.

Signed-off-by: Joakim Andre Tønnesen <joakim.tonnesen@nordicsemi.no>

### CODE REVIEW

File: drivers/lte_link_control/Kconfig [hunk 1]
```diff
@@ -11,6 +11,10 @@ menuconfig LTE_LINK_CONTROL
 
 if LTE_LINK_CONTROL
 
+config LTE_LINK_CONTROL_LOG_LEVEL
```
Comment (rlubos) [inline drivers/lte_link_control/Kconfig]: You could use logger template here, just as other modules do:
https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/5f577bbb2f40d0f0f83c3a860692aa029068c812/subsys/net/lib/nrf_cloud/Kconfig#L48

File: drivers/lte_link_control/Kconfig [hunk 2]
```diff
@@ -69,6 +73,23 @@ config LTE_EDRX_REQ_VALUE
 		Extended DRX parameters information element.
 		See 3GPP TS 24.008, subclause 10.5.5.32.
 
+config LTE_LEGACY_PCO_MODE
+	bool "Enable legacy LTE Protocol Configuration Options mode"
+	default n
```
Comment (rlubos) [inline drivers/lte_link_control/Kconfig]: `n` is default, no need to explicitly add it.

File: drivers/lte_link_control/Kconfig [hunk 3]
```diff
@@ -69,6 +73,23 @@ config LTE_EDRX_REQ_VALUE
...
+
+config LTE_PDP_CMD
+	bool "Enable Packet Data Protocol AT command"
+	default n
```
Comment (rlubos) [inline drivers/lte_link_control/Kconfig]: Same here, `n` is default by definition.

File: drivers/lte_link_control/lte_lc.c [hunk 4]
```diff
@@ -126,6 +149,7 @@ static int w_lte_lc_init_and_connect(struct device *unused)
 				break;
 			}
 		}
+		k_sleep(K_MSEC(10));
```
Comment (rlubos) [inline drivers/lte_link_control/lte_lc.c]: Why is this needed? Is blocking `recv` misbehaving?

Comment (joakimtoe) [reply drivers/lte_link_control/lte_lc.c]: yes, this is related to missing implementation of bsd_os_timedwait.

CC @jtguggedal

Comment (rlubos) [reply drivers/lte_link_control/lte_lc.c]: Ok, makes sense, thanks for claryfing

File: drivers/lte_link_control/lte_lc.c [hunk 5]
```diff
@@ -106,6 +113,22 @@ static int w_lte_lc_init_and_connect(struct device *unused)
...
 		return err;
 	}
+#endif
+#if defined(CONFIG_LTE_LEGACY_PCO_MODE)
+	err = at_cmd(at_socket_fd, legacy_pco, AT_CMD_SIZE(legacy_pco));
+	if (err) {
+		close(at_socket_fd);
+		return err;
+	}
+	LOG_INF("Using legacy LTE PCO mode...\n");
```
Comment (jtguggedal) [inline drivers/lte_link_control/lte_lc.c]: The `\n` is redundant here, as logger module adds a newline as well, same other places in the file

Comment (lemrey) [reply drivers/lte_link_control/lte_lc.c]: @joakimtoe Let's fix this and merge it.

Comment (joakimtoe) [reply drivers/lte_link_control/lte_lc.c]: @lemrey done!
